pkg/executor/wasm/executor.go

```diff
@@ -57,8 +57,17 @@ func (*Executor) ShouldBidBasedOnUsage(
 	return resource.NewChainedResourceBidStrategy().ShouldBidBasedOnUsage(ctx, request, usage)
 }
 
+// Wazero: is compliant to WebAssembly Core Specification 1.0 and 2.0.
+//
+// WebAssembly1:  linear memory objects have sizes measured in pages. Each page is 65536 (2^16) bytes.
+// In WebAssembly version 1, a linear memory can have at most 65536 pages, for a total of 2^32 bytes (4 gibibytes).
+
+const WASM_ARCH = 32
+const WASM_PAGE_SIZE = 1 << (WASM_ARCH / 2)
```

Comment:
I'm not sure this is based on the arch size, it's entirely possible they'll stick with 65536 in wasm64, so we should go with 65536 for now.

Reply:
<img width="561" alt="image" src="https://github.com/bacalhau-project/bacalhau/assets/24226219/a7bd2238-29b1-408d-a455-f93cfd9b1312">

65536 seemed like magic number to me, so that's why put it up.

Reply:
I agree, so we can put it into a const as 65536, but it's in the spec and so we should invent ways to generate the number if it isn't necessary.

Reply:
WASM_ARCH isn't really a magic number per say, it implies 32 bit arch. But i get ur point, sure the doc would do

Reply:
FWIW, it was only the PAGE_SIZE I meant, calculating the limit was fine, but don't change it now we can fix it up when wasm64 arrives.  In the meantime, need to figure out this apparently python error in CircleCI.

Reply:
@rossjones i don't mind fixing it. done